### ISSUE TITLE
Shows wrong data . verts are to high

### ISSUE THREAD

**Issue by schroef** [Vinc3r/Polycount#25]
v2020.10.27
bl2.83

I just now noticed you showing wrong vert count with high poly models. When i tested you test scene from this repo. The object "un-miyion-oui-monsieur" show much less data than you addons shows.

See below the screengrabs. I think that 'k" at the end should be moved left
![Screen Shot 2021-03-02 at 17 18 33](https://user-images.githubusercontent.com/6923008/109716959-d6cec100-7b7b-11eb-85fd-74deb4c23cd5.png)
![Screen Shot 2021-03-02 at 17 18 35](https://user-images.githubusercontent.com/6923008/109716962-d6cec100-7b7b-11eb-906e-f02cec91ff31.png)

The model has the following vertices
1.048.577 verts (blender count)
1.048.57k verts (polycount addon)

Notice the difference according to your count this models should be 10.485.700k

you basicall added k value for and didnt subtract 2digits

ooorrr...

im reading it wrong. Perhaps due to the dots its a but misleading. I guess its the last case. Due to the many dots it looks different. I guess using different dividers should help. Perhaps floating '

So something like 1'048k

**Comment by Vinc3r** [comment]
yes your last point is the right one: I'm french maybe I have the wrong way of using decimal character "."?
for example I write 2048 or 2 048, for decimal I write 20,48 (identical to 20.48 in code)

so, is `1'048.56k` seems right to you? actually it's a semantic question :)

**Comment by Vinc3r** [comment]
ok, i think i will follow Blender convention, so writing `1,048.56k` (equal to 1,048,560) is the way to do, or maybe better, when "k" option is used, i may disable decimals, giving `1,049k` like you suggested 👍🏾

or just keep one digit, like Twitter does, giving `1,048.6k` 
![image](https://user-images.githubusercontent.com/3873421/109777396-f6113100-7c03-11eb-9a0a-fe4e94089b8d.png)

**Comment by Vinc3r** [comment]
https://en.wikipedia.org/wiki/Decimal_separator#Examples_of_use 😅

**Comment by schroef** [comment]
Yeah its kinda of fuzzy if you see how many different methods there are. I think the point is, is showing those 2 extra decimals it doesnt really matter that much. I mean if you see a big number and its is in K's you kinda need to think about what the actual number is right. I mean with bigger numbers. Perhap if you drop those decimals. 

So for instance 2080770 is not 2.088.77k better round that to 2.088k. The thing which is confusing to me is that the dot also also a 1000 seperator but also a decimal. To me at least this is confusing.

![1000-separator-confusing-1](https://user-images.githubusercontent.com/6923008/109865204-2b387600-7c3a-11eb-9305-a8e3a30087bb.png)
![1000-separator-confusing-2](https://user-images.githubusercontent.com/6923008/109865206-2bd10c80-7c3a-11eb-974e-94f29364b162.png)

really wondering why blenders internal counter is so super fast. But they use C++ internal code probably. This BM mesh makes it kinda slow. I also noticed that other addon make the GUI super slow, he's constantly do panel refreshes and doing lots of calculations, not reallu ideal! I tried a spehere which boosted to 2mil verts, i got an old machine, but my GUI started to bug down because of his panel design

**Comment by schroef** [comment]
I thought i found a method to make it work instantly using mesh_eval. Sadly this can only pick the vertices and not the tris and area. I also tested a method using scene.statistics() and then stripping that. But it wont show the data properly when i run it with an operator. It doesnt update the scene in between. My idea was to hide every object mesh then run this statistic call. Because in blender it will only show the data of visible objects.

**Comment by Vinc3r** [comment]
thanks for all your investigations :)
yea in first versions of this addon I was updating stats on every user action, but it was too heavy. I have already in mind to enable user to get automatic refresh, but using a kind of timer and not on every draw, cf #3 
About the mesh_eval, if it's allow only vertex count it's not convenient.
About default Blender statistics, your idea could be interesting. However in huge scene, simulation hide/show on every object could probably be lagging too.

**Comment by schroef** [comment]
Well the show/hide is super fast, but indeed, when its like 100 or 1000s of object it would slow. But that would make this addon as it currently is also slow. For me now its already slow using 2 meshes and around 2 million vertices. Thats not good!

I think it would be better if it only worked on selection, dont you?

I meanif nothing is selected, then there's already Blender default counter.

**Comment by Vinc3r** [comment]
sorry for the delay, i haven't enough of spare time these days.

i will think about the show/hide option, it can mean a total rewriting.

> I mean if nothing is selected, then there's already Blender default counter.

i still want my addon to show per-object stats in a glance

**Comment by schroef** [comment]
Yeah it's nice yet very slow, especially when scene size is big.

The current method is much easier. Simply hide all other objects.
Isn't that an idea? Make a panel which hides everything see we do see the
current stats. That way you don't need to do the bmesh thing which is
slowing down blender immense
